tests/datatypes/test_str.py

```diff
@@ -482,6 +500,90 @@ def test_bogus_specifier(self, js_cleaner, py_cleaner):
 
             self.assertCodeExecution(test, js_cleaner = js_cleaner, py_cleaner = py_cleaner)
 
+        @transforms(
+            js_bool = False,
+            decimal = False,
+            float_exp = False,
+            memory_ref = False
+        )
+        def test_with_kwargs(self, js_cleaner, py_cleaner):
+
+            tests = adjust("""
+                print(">>> 'format this: %(arg)s' % {'arg':'spam'}")
+                print('format this: %(arg)s' % {'arg':'spam'})
+                print(">>> 'format this: %(arg)s' % {'arg':'spam', 'arg2':'eggs'}")
+                print('format this: %(arg)s' % {'arg':'spam', 'arg2':'eggs'})
+                print('Done.')
+                """)
+
+            self.assertCodeExecution(tests, js_cleaner = js_cleaner, py_cleaner = py_cleaner)
+
+        @transforms(
+            js_bool = False,
+            decimal = False,
+            float_exp = False,
+            memory_ref = False
+        )
+        def test_with_tuple_of_dicts(self, js_cleaner, py_cleaner):
+
+            tests = adjust("""
+                print(">>> 'format this: %(arg)s' % ({'arg':'spam'}, {'arg2':'eggs'})")
+                print('format this: %(arg)s' % ({'arg':'spam'}, {'arg2':'eggs'}))
+                print('Done.')
+                """)
+
+            self.assertCodeExecution(tests, js_cleaner = js_cleaner, py_cleaner = py_cleaner)
+
+        @transforms(
+            js_bool = False,
+            decimal = False,
+            float_exp = False,
+            memory_ref = False
+        )
+        def test_with_kwargs(self, js_cleaner, py_cleaner):
+
+            tests = adjust("""
+                print(">>> 'format this: %(arg' % {'arg':'spam'}")
+                print('format this: %(arg' % {'arg':'spam'})
+                print('Done.')
+                """)
+
+            self.assertCodeExecution(tests, js_cleaner = js_cleaner, py_cleaner = py_cleaner)
+
+        @transforms(
+            js_bool = False,
+            decimal = False,
+            float_exp = False,
+            memory_ref = False
+        )
+        def test_wildcard_with_kwargs(self, js_cleaner, py_cleaner):
+
+            tests = adjust("""
+                print(">>> '%*(spam)s' % {'spam': 'eggs'}")
+                print('%*s' % {'spam': 'eggs'})
```

Comment:
Not sure which of these you're intending to test here, but the test case doesn't match the debug line.